lib/active_publisher/async/in_memory_adapter/consumer_thread.rb

```diff
@@ -41,10 +42,19 @@ def make_channel
           channel
         end
 
+        def send_heartbeat
+          return unless @heartbeats.empty?
+          @heartbeats << :hello
+        end
+
         def start_thread
           return if alive?
           @thread = ::Thread.new do
             loop do
+              # If the queue is empty, we should continue to send heartbeats to the consumer.
+              send_heartbeat
+              next sleep 0.1 if queue.empty?
```

Comment:
this is optional, but if you want to remove the sleep we could merge in this: https://github.com/abrandoned/multi_op_queue/pull/1

then below

```
current_messages = queue.pop_up_to(50)
send_heartbeat
next unless current_messages
```

Reply:
like the idea of incorporating this into the `pop_up_to` call

Reply:
I agree! Very nice.